test/dns-server.test.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  createHostedZone,
+  resetHostedZone,
+  createRecord,
+  updateRecord,
+  deleteRecord,
+  listRecords,
+  getChangeStatus,
+  RecordType,
+} from "../app/lib/dns-server";
+
+describe("DNS server lib function test", () => {
+  let hostedZoneId: string;
+
+  beforeAll(async () => {
+    hostedZoneId = await createHostedZone("starchart.com");
+  });
+
+  afterAll(() => resetHostedZone(hostedZoneId));
+
+  test("Hosted zone is created and hosted zone is returned", () => {
+    expect(hostedZoneId.length).toBeGreaterThan(0);
+    expect(/^[A-Z0-9]+/.test(hostedZoneId)).toBe(true);
```

Comment:
I've never seen clear docs on the format of this string.  I'd suggest what you do with the length is enough.  Route53/moto could change the format in the future.  We don't really care what it is, just that we get something back.